File: CSharp/ImageProcessing/ComSaveFileDialog.cs
using Microsoft.Win32;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ImageProcessing
{
    class ComSaveFileDialog : ComFileDialog
    {
        private Stream m_stream;
        private bool m_bCreateStream;

        public Stream Stream
        {
            get { return m_stream; }
        }

        public bool CreateStream
        {
            set { m_bCreateStream = value; }
            get { return m_bCreateStream; }
        }

        public ComSaveFileDialog(bool _bCreateStream = false) : base()
        {
            m_bCreateStream = _bCreateStream;
        }

        ~ComSaveFileDialog()
        {
        }

        public override bool ShowDialog()
        {
            bool bRst = false;

            SaveFileDialog saveDialog = new SaveFileDialog();
            saveDialog.FileName = base.FileName;
            saveDialog.InitialDirectory = base.InitialDirectory;
            saveDialog.Filter = base.Filter;
            saveDialog.FilterIndex = base.FilterIndex;
            saveDialog.Title = base.Title;
            saveDialog.CheckFileExists = base.CheckFileExists;
            saveDialog.CheckPathExists = base.CheckPathExists;
            if (saveDialog.ShowDialog() == true)
            {
                base.m_strFilePass = saveDialog.FileName;
                try
                {
                    if (m_bCreateStream)
                    {
                        m_stream = saveDialog.OpenFile();
                    }
                }
                catch(Exception)
                {
                    bRst = false;
                    return bRst;
                }
                bRst = true;
            }

            return bRst;
        }
    }
}

File: Vb/ImageProcessing/ComSaveFileDialog.vb
Imports Microsoft.Win32

Namespace ImageProcessing
    Public Class ComSaveFileDialog : Inherits ComFileDialog
        Public Sub New()
            MyBase.New()
        End Sub

        Protected Overloads Sub Finalize()
            MyBase.Finalize()
        End Sub

        Public Overrides Function ShowDialog() As Boolean
            Dim bRst As Boolean = False
            Dim saveDialog As SaveFileDialog = New SaveFileDialog()
            saveDialog.FileName = MyBase.FileName
            saveDialog.InitialDirectory = MyBase.InitialDirectory
            saveDialog.Filter = MyBase.Filter
            saveDialog.FilterIndex = MyBase.FilterIndex
            saveDialog.Title = MyBase.Title
            saveDialog.CheckFileExists = MyBase.CheckFileExists
            saveDialog.CheckPathExists = MyBase.CheckPathExists
            If (saveDialog.ShowDialog() = True) Then
                MyBase.m_strFilePass = saveDialog.FileName
                bRst = True
            End If

            Return bRst
        End Function
    End Class
End Namespace
